compliance/repository/src/test/java/org/eclipse/rdf4j/repository/sparql/federation/RepositoryFederatedServiceIntegrationTest.java

```diff
@@ -262,6 +265,59 @@ public void test8a_subSelectAll() throws Exception {
 		assertResultEquals(evaluateQuery(query), "var", Lists.newArrayList(l("val1")));
 	}
 
+	@Test
+	public void test9_connectionHandling() throws Exception {
+
+		System.setProperty("org.eclipse.rdf4j.repository.debug", "true");
+		List<Value> values = Lists.newArrayList();
+		for (int i = 0; i < 10; i++) {
+			values.add(l("value" + i));
+		}
+		addData(serviceRepo,
+				values.stream()
+						.map(value -> vf.createStatement(iri("s1"), RDFS.LABEL, value))
+						.collect(Collectors.toList()));
+
+		ExecutorService executor = Executors.newFixedThreadPool(5);
+		try {
+			for (int i = 0; i < 5; i++) {
+				executor.submit(() -> {
+
+					String query = "SELECT ?var WHERE { SERVICE <urn:dummy> { ?s ?p ?var  } }";
+					assertResultEquals(evaluateQuery(query), "var", values);
+				});
+			}
+
+		} finally {
+			executor.shutdown();
+			executor.awaitTermination(10, TimeUnit.SECONDS);
+		}
+
+	}
+
+	@Test
+	public void test10_consumePartially() throws Exception {
+
+		List<Value> values = Lists.newArrayList();
+		for (int i = 0; i < 10; i++) {
+			values.add(l("value" + i));
+		}
+		addData(serviceRepo,
+				values.stream()
+						.map(value -> vf.createStatement(iri("s1"), RDFS.LABEL, value))
+						.collect(Collectors.toList()));
+
+		String query = "SELECT ?var WHERE { SERVICE <urn:dummy> { ?s ?p ?var  } }";
+		try (RepositoryConnection conn = localRepo.getConnection()) {
+			try (TupleQueryResult tqr = conn.prepareTupleQuery(query).evaluate()) {
+
+				// consume only two items
+				tqr.next();
+				tqr.next();
```

Comment:
Could you add a comment here that explains why this test is a good indicator if things are properly closed? I know you explained in the PR description but I'm thinking it would be good to have it as a code comment.

Reply:
Added a comment to both tests.